### PR TITLE
fix(monitoting): bump storage up

### PR DESCRIPTION
**Pull request by georgettica**
according to internal consulting, we want the storage to hold for 15d without filling up

### CODE REVIEW

File: deploy/cluster-monitoring-config/cluster-monitoring-config.yaml
Comment (jewzaam) [inline deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: With the bump in both storage and retention we are not gaining any headroom on consuming all storage for prometheus, which is the goal with OSD-4875.  Recommend we do not make this change in retention and leave the PR to only the storage change.

```suggestion
      retention: 7d
```

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: deploy/cluster-monitoring-config/cluster-monitoring-config.yaml
Comment (NautiluX) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: I think @jewzaam suggests to change both, just not in the same PR/promotion

Comment (jewzaam) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: I don't see why we need the retention.  What is the driver to increase retention?

Comment (NautiluX) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: retention has been reduced from 10 to 7 days as a short-term solution to get rid of the alerts. Now we want to increase the retention instead as 7 days of metrics is a short period. With OSD-5221 we did a spike to figure out the right way to go forward with this, got approval for enough storage to hold 15 days, and that's what @georgettica is implementing here. Current config has enough headroom for 7 days, so the assumption is the higher amount should be sufficient for 15 days.

Comment (jewzaam) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: Thought we were sitll hitting storage problems with 7d retention and 50Gi storage, so I disagree that this bump to 14d retention and 100Gi storage fixes the storage issue.  If we want to bump to 10d I am OK with that.

Comment (georgettica) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: What size would be okay on your end? You can see the calculations in the spike we did and devise from there.
I would like to work on both avenues (bigger storage for 14 days or current suggestion for 10) in case one gets blocked (by BU or internally)

Comment (jewzaam) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: Without a use case driving 14d over 10d I'm +1 to 100Gi w/ 10d.  Having over a week is nice and I have felt 10d was a reasonable number after we dropped it from 14d originally.

Comment (RiRa12621) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: I think I commented on the ticket: 2 weeks is the minimum that allows you to do WoW for any day in the last seven days.
For example if you need to compare the behavior of a cluster between two weeks.
As long as we don't federate all metrics off cluster, I think we shouldn't lose that ability on cluster, hence the 14d

Comment (jewzaam) [reply deploy/cluster-monitoring-config/cluster-monitoring-config.yaml:15]: I read back through OSD-5221 reasoning.  And considering a low volume of KubePersistentVolumeFullInFourDays alerts showing in Tableau, no hits in telemeter in the recent past, inability to get a result from telemeter for longer time slices (just get a timeout for queries), we're probably OK with the 7d and we can adjust retention down again if necessarily I'll approve.

**Comment by georgettica** [comment]
/hold
a valid concen was given by @jewzaam , need to test before merging

**Comment by RiRa12621** [comment]
Afaik the operator cannot adjust that and we need to manually do it.
/cc s-urbaniak @lilic 
Can you confirm that?

**Comment by s-urbaniak** [comment]
we have no automatic data migration in place if the PVC size changes. If the underlying provider does not support volume expansion internally data needs to be moved manually.

**Comment by georgettica** [comment]
@jewzaam I found out how to do the operation on one cluster.. do you have an idea on how to make this run on all of the fleet?

**Comment by jewzaam** [comment]
@georgettica you can create a cronjob to deploy across the fleet.  There's an SOP on pruning resources but it's the same basic idea: https://github.com/openshift/ops-sop/blob/master/v4/howto/prune-resources.md

**Comment by georgettica** [comment]
@jewzaam the job has been found to not crash and burn, and run the patches.. now I will try to:
- stop syncset
- change prom size and retention
- run the job

and see if it works

**Comment by georgettica** [comment]
@s-urbaniak @jewzaam @RiRa12621 I added a command to manually increase the storage and verified all of the pvc had an option to be scaled up

**Comment by georgettica** [comment]
seems like there is a bug in https://github.com/openshift/managed-cluster-config/blob/796398a7937ecab8496951e11c1c1eaf0138cb7c/scripts/generate_template.py#L79

seems that the `oc patch` is being recognized as a a problem

**Comment by georgettica** [comment]
that wasn't the issue, solved in https://github.com/openshift/managed-cluster-config/pull/559

**Comment by georgettica** [comment]
@jewzaam, it seems that it's fine now.. so I guess it can get a +1?

**Pull request merged**
